engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreSubmitActor.scala

```diff
@@ -87,7 +87,13 @@ final case class WorkflowStoreSubmitActor(store: WorkflowStore, serviceRegistryA
         MetadataEvent(MetadataKey(id, None, WorkflowMetadataKeys.SubmissionSection, WorkflowMetadataKeys.SubmissionSection_Options), MetadataValue(sourceFiles.workflowOptionsJson))
       )
 
-      serviceRegistryActor ! PutMetadataAction(submissionEvents)
+      // Don't publish metadata for either workflow type or workflow type version if not defined.
+      val workflowTypeAndVersionEvents: List[Option[MetadataEvent]] = List(
+        sourceFiles.workflowType map { wt => MetadataEvent(MetadataKey(id, None, WorkflowMetadataKeys.SubmissionSection, WorkflowMetadataKeys.SubmissionSection_WorkflowType), MetadataValue(wt)) },
```

Comment:
There may not have been a workflow type specified in the submission.  Ultimately we'll need to default to a workflow type if none is specified, but not at this layer.